pageserver/src/bin/layer_map_analyzer.rs

```diff
@@ -0,0 +1,212 @@
+use anyhow::Result;
```

Comment:
Top-level docstring would be useful. Suggestion:

"
Tool for extracting content-dependent metadata about layers. Useful for scanning real project layer files and evaluating the effectiveness of different heuristics on them.

Currently it only analyzes holes, which are regions within the layer range that the layer contains no updates for. In the future it might do more analysis (maybe key quantiles?) but it should never return sensitive data.
"

Also maybe `layer_metadata_analyzer` is another alternative name. The current name makes it sound like the input should be a layer map. Your choice.

Reply:
1. Thank you for writing this description for me. Added.
2. Right now holes is not part of layer metadata. It will be if #3348 will be merged.
3.